src/sentry/api/serializers/rest_framework/dashboard.py

```diff
@@ -490,7 +491,23 @@ def validate(self, data):
             and data.get("limit") is None
             and has_columns
         ):
-            raise serializers.ValidationError({"limit": "limit is required."})
+            defaultLimit = 5
+            maxLimit = 10
+            numQueries = len(data.get("queries", []))
+            numYAxes = len(data.get("queries", [])[0].get("aggregates", []))
+            limit = defaultLimit
+
+            if numQueries == 0 or numYAxes == 0:
+                limit = defaultLimit
+            else:
+                limit = floor(maxLimit / (numQueries * numYAxes))
+
+            sentry_sdk.capture_message(
+                f"Dashboard Widget limit was not set. Suggested maximum limit is {limit}."
+            )
+            raise serializers.ValidationError(
+                {"limit": f"limit is required. the maximum limit is ${limit}."}
```

Comment:
```suggestion
                {"limit": f"limit is required. The maximum limit is ${limit}."}
```
nit: Since we're using sentences to phrase the error here, just figured the second sentence should be capitalized